x-pack/plugins/stack_alerts/server/rule_types/es_query/lib/fetch_search_source_query.ts

```diff
@@ -61,7 +73,19 @@ export async function fetchSearchSourceQuery({
 
   const searchResult = await searchSource.fetch();
 
+  const link = await generateLink(
+    initialSearchSource,
+    services.share.url.locators.get<DiscoverAppLocatorParams>('DISCOVER_APP_LOCATOR')!,
+    services.dataViews,
+    index,
+    dateStart,
```

Comment:
So here's the hopefully last thing I've found. When using `dataStart` to generate the link, this doesn't take 
`excludeHitsFromPreviousRun` into consideration (L120)

`updateSearchSource` on L95 returns `dataStart` always the full time range. But given `excludeHitsFromPreviousRun` is set, there can be an additional filter to prevent to have documents counted twice. This leads to the following mismatch of count when clicking the link in the notification:

![Discover_-_Elastic_und_kibana-alerting-slack-tests_-_Elastic_-_4_new_items_-_Slack_und__Discover__Alerting__Use_Discover_locator_for_alert_results_link_by_dimaanj_·_Pull_Request__146403_·_elastic_kibana](https://user-images.githubusercontent.com/463851/209971296-226344fd-db4e-442c-bcec-511ea3132c24.png)
Correct me if I'm wrong but I think this was already broken before this PR?

Reply:
ok, this ain't an issue of this PR, just tried out on a previous version, works the same ... we should open an issue for this

Reply:
Created #148282